lib/connect.js

```diff
@@ -731,11 +731,16 @@ regular).
      * @param {Array} [segment] Filter instruments based on exchange (NSE, BSE, NFO, BFO, CDS, MCX).
      * If no `segment` is specified, all instruments are returned.
      */
-    self.getInstruments = function(exchange) {
-        if(exchange) {
+    self.getInstruments = function(exchange, segment) {
+        if(exchange && !segment) {
             return _get("market.instruments", {
                 "exchange": exchange
             }, null, transformInstrumentsResponse);
+        } else if (exchange && segment ) {
+            return _get("market.instruments", {
+                "exchange": exchange,
+                "segment": segment
```

Comment:
Adding `segment` to query param won't filter by segment since `market.instruments` endpoint doesn't support `segment` in URL.